### ISSUE TITLE
Improve Documentation

### ISSUE THREAD

**Issue by guenthermi** [jina-ai/finetuner#523]
Implement the feedback we got for the documentation:
- [ ] better describe good parameter settings (e.g. learning rate, number of epochs etc.), and encourage users to try out different configurations if fine-tuning is not working properly
- [ ] document `select_model` attribute of the Finetuner Executor
- [ ] Add documentation about "finetuner-label" in https://finetuner.jina.ai/walkthrough/create-training-data/ 
- [ ] Explain task management, i.e., explain the notion of experiments, runs, and their names in more detail (e.g. also describe how they are automatically set when no names are provided) 
- [ ] Explain model management, i.e., hubble artifacts vs local model files, why downloading and re-opening, ...
